modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -56,8 +93,67 @@ CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
     _inputs(getParam<std::vector<MeshGeneratorName>>("inputs")),
     _empty_key(getParam<MeshGeneratorName>("dummy_assembly_name")),
     _pattern(getParam<std::vector<std::vector<unsigned int>>>("pattern")),
-    _extrude(getParam<bool>("extrude"))
+    _extrude(getParam<bool>("extrude")),
+    _mesh_periphery(getParam<bool>("mesh_periphery")),
+    _periphery_meshgenerator(getParam<MooseEnum>("periphery_generator")),
+    _periphery_region_id(isParamValid("periphery_region_id")
+                             ? getParam<subdomain_id_type>("periphery_region_id")
+                             : -1),
+    _outer_circle_radius(isParamValid("outer_circle_radius") ? getParam<Real>("outer_circle_radius")
+                                                             : 0),
+    _outer_circle_num_segments(isParamValid("outer_circle_num_segments")
+                                   ? getParam<unsigned int>("outer_circle_num_segments")
+                                   : 0),
+    _periphery_block_name(isParamValid("periphery_block_name")
+                              ? getParam<std::string>("periphery_block_name")
+                              : (std::string) "outer_core"),
+    _periphery_num_layers(
+        isParamValid("periphery_num_layers") ? getParam<unsigned int>("periphery_num_layers") : 1),
```

Comment:
some of these could be moved to defaults
then instead of isParamValid below, you would use isParamSetByUser 

the advantages are:
- a shorter constructor
- some of these could just actually have defaults
num_layers of 1 or block_name of outer_core are good defaults

Reply:
https://mooseframework.inl.gov/source/meshgenerators/PeripheralRingMeshGenerator.html does have defaults